model/common/src/icon4py/model/common/decomposition/mpi_decomposition.py

```diff
@@ -210,7 +211,7 @@ def _slice_field_based_on_dim(self, field: gtx.Field, dim: gtx.Dimension) -> dat
         else:
             raise ValueError(f"Unknown dimension {dim}")
 
-    def exchange(self, dim: gtx.Dimension, *fields: Sequence[gtx.Field]):
+    def exchange(self, dim: gtx.Dimension, *fields: Sequence[gtx.Field]) -> MultiNodeResult:
```

Comment:
```suggestion
    def exchange(self, dim: gtx.Dimension, *fields: gtx.Field) -> MultiNodeResult:
```
then the 2 changes below can probably be reverted